### PR TITLE
Apisetup Book Setup

### PR DESCRIPTION
**Pull request by gjuliao**
## Requirements

1. Make calls to fetch with Axios
2. Installed middleware
3. Refactor Add Book and Remove Book reducer functions and actions.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,45 +1,67 @@
...
-    title: 'I will die in a foreign land',
-  },
-];
-
-const bookReducer = (state = initialState, action) => {
-  switch (action.type) {
-    case ADD_BOOK: { return [...state, action.book]; }
-    case REMOVE_BOOK: return state.filter((book) => book.id !== action.id);
-    default: return state;
+import axios from '../axiosInstance';
```
Comment (Whoistolu) [inline src/redux/books/books.js:2]: - [ ] [REQUIRED] As stated in the project requirements. Kindly ensure that you use `createAsyncThunk` to create thunk action creators. While using that, you will need to 
`import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';`
and also import axios just as you have done.

Here is a typical example:
```
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios'; 


export const getBooks = createAsyncThunk(
  'books/getBooks',
  async () => {
    const response = await axios.get(
      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/',
    );
    return response.data;
  },
); 
```

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @gjuliao 

Good job so far, your project works fine. The add and remove feature looks good!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._
- [ ] [REQUIRED] Kindly ensure you properly license this project. Your code is a creative work, and like most creative works, it needs copyright protection. It is now a core requirement to properly license your code.  Kindly use this [template](https://choosealicense.com/licenses/mit/).

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gjuliao** [comment]
Note: MIT license was added in the dev branch directly.

**Review by tugsem — APPROVED** [review]
## STATUS: APPROVED 👍🎯  
Hi @gjuliao , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⭐ 
✔️ Well done 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**